lib/active_fedora/schema_indexing_strategy.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveFedora
+  ##
+  # An application strategy to also apply solr indexes for each property.
+  # @note If how a field is indexed changes based on property, this would be a
+  #   good place to define that.
+  class SchemaIndexingStrategy
+    # @param [Array<Symbol>] index_types Ways to index all properties.
+    def initialize(index_types=[])
```

Comment:
Instead of passing index_types, how about passing in a class or a block that does the indexing? 
```ruby
class BookIndexer
   # returns a block that applies indexes
   def index
     lambda { |index| index.as(:symbol) }
   end
end
apply_schema BookSchema, SchemaIndexingStragegy.new(BookIndexer)
```
Then apply() can pass BookIndexer.new.index as the third argument to object.property?